packages/report-flaky-tests/tsconfig.json

```diff
@@ -0,0 +1,15 @@
+{
+	"extends": "../../tsconfig.base.json",
+	"compilerOptions": {
+		"module": "CommonJS",
+		"declarationDir": "build-types",
+		"rootDir": "lib",
```

Comment:
Why `lib`? In most other packages it's `src`.

Reply:
The build steps (`npm run dev` and `npm run build`) will automatically transpile everything in `src` into `build` and `build-module` with Babel. However, it doesn't make sense to compile GH Action into ESM because that's not supported. I guess we can still use `src` though if we want to make it consistent.

Reply:
Do you have a preference @gziolo?

Reply:
The `lib` naming convention is copied from the `project-management-automation` package.

Reply:
In other packages we use `lib` when we don't want them to be transpiled with Babel, `@wordpress/prettier-config` and similar non-production packages would be a good example. Here, we have TypeScript files so the code won't work out of the box. That's also why `@wordpress/project-management-automation` uses JSDoc to import TypeScript types to avoid the additional step of removing TypeScript annotations from the source code.

> However, it doesn't make sense to compile GH Action into ESM because that's not supported. 

Node 14 is the minimum LTS version and it fully supports ESM. What is the reason it doesn't work with GitHub actions?

The build process we have configured produces also CommonJS version if necessary. For production packages, it's the other way around, we use only ESM and produce CommonJS for optional 3rd party usage.

Reply:
> Node 14 is the minimum LTS version and it fully supports ESM. What is the reason it doesn't work with GitHub actions?

Oh, wait. Seems like GH Actions support [node 16](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsusing) now. The last time I tried it only supported node 12.

> The build process we have configured produces also CommonJS version if necessary. For production packages, it's the other way around, we use only ESM and produce CommonJS for optional 3rd party usage.

For a private and internal package like this, building it into two different formats still isn't necessary, but that's a different issue. We could keep it simple and still use the default `src` and let Babel compiles it.

Reply:
I'd love to have the monorepo have individual build steps for each package defined in its `package.json`. So that we don't have to deal with this weird naming convention setting. A bonus benefit is that we can build each package separately. Like in the flaky tests reporter workflow, we don't have to build the whole project, but only the `report-flaky-tests` package. I don't think currently this can be easily done?

Reply:
Think I'm leaning towards `src` just because `lib` implies that no compilation is required whatsoever which isn't true here (it requires `tsc`).

> I don't think currently this can be easily done?

I suppose you you could add a script to the `package.json` in `report-flaky-tests` and `cd report-flaky-tests; npm run ...`? It's a bit obtuse, I probably wouldn't recommend it.

Reply:
> Think I'm leaning towards `src` just because `lib` implies that no compilation is required whatsoever which isn't true here (it requires `tsc`).

I've changed it to `src` in https://github.com/WordPress/gutenberg/pull/41106/commits/9c451ee01860b5299ff73258df37804e77f5a104 👍

> I suppose you you could add a script to the `package.json` in `report-flaky-tests` and `cd report-flaky-tests; npm run ...`? It's a bit obtuse, I probably wouldn't recommend it.

I guess that could work? But then I _think_ we have to also move some dev dependencies there (`babel` or `tsc` for instance). Either way, the built result will still be in the `build` folder, which conflicts with the default result when the code is in `src`. I guess that would be a little weird 😅.

Reply:
I think it could get confusing, let's stick to having scripts at the top level only.